pkg/antctl/antctl.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package antctl
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/server/mux"
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/antctl/handlers"
+	"github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned/scheme"
+	"github.com/vmware-tanzu/antrea/pkg/monitor"
+	"github.com/vmware-tanzu/antrea/pkg/version"
+)
+
+// unixDomainSockAddr is the address for antctl server in local mode.
+const unixDomainSockAddr = "/var/run/antctl.sock"
+
+type versionResponse struct {
+	handlers.ComponentVersionResponse `json:",inline" yaml:",inline"`
+	AntctlVersion                     string `json:"antctlVersion" yaml:"antctlVersion"`
```

Comment:
To make it more explicit, consider name the struct transformedVersionResponse.

Reply:
Done.